tests/cli/test_flow_run.py

```diff
@@ -275,3 +277,98 @@ def test_wrong_id_exits_with_error(self):
             expected_code=1,
             expected_output_contains=f"Flow run '{bad_id}' not found!\n",
         )
+
+
+class TestFlowRunLogs:
+    PAGE_SIZE = 200
+
+    @pytest.mark.parametrize("state", [Completed, Failed, Crashed, Cancelled])
+    async def test_when_less_than_page_size_logs_exists_then_no_pagination(
+        self, orion_client, state
+    ):
+        # Given
+        flow_run = await orion_client.create_flow_run(
+            name="scheduled_flow_run", flow=hello_flow, state=state()
+        )
+
+        # Create less than page size flow run logs
+        logs = [
+            LogCreate(
+                name="prefect.flow_runs",
+                level=20,
+                message=f"Log {i} from flow_run {flow_run.id}.",
+                timestamp=datetime.now(tz=timezone.utc),
+                flow_run_id=flow_run.id,
+            )
+            for i in range(self.PAGE_SIZE - 1)
+        ]
+        await orion_client.create_logs(logs)
+
+        # When/Then
+        await run_sync_in_worker_thread(
+            invoke_and_assert,
+            command=[
+                "flow-run",
+                "logs",
+                str(flow_run.id),
+            ],
+            expected_code=0,
+            expected_output_contains=[
+                f"Flow run '{flow_run.name}' - Log {i} from flow_run {flow_run.id}."
+                for i in range(self.PAGE_SIZE - 1)
+            ],
+        )
+
+    @pytest.mark.parametrize("state", [Completed, Failed, Crashed, Cancelled])
+    async def test_when_more_than_page_size_logs_exists_then_pagination(
+        self, orion_client, state
+    ):
+        # Given
+        flow_run = await orion_client.create_flow_run(
+            name="scheduled_flow_run", flow=hello_flow, state=state()
+        )
+
+        # Create more than page size flow run logs
+        logs = [
+            LogCreate(
+                name="prefect.flow_runs",
+                level=20,
+                message=f"Log {i} from flow_run {flow_run.id}.",
+                timestamp=datetime.now(tz=timezone.utc),
+                flow_run_id=flow_run.id,
+            )
+            for i in range(self.PAGE_SIZE + 1)
+        ]
+        await orion_client.create_logs(logs)
+
+        # When/Then
+        await run_sync_in_worker_thread(
+            invoke_and_assert,
+            command=[
+                "flow-run",
+                "logs",
+                str(flow_run.id),
+            ],
+            expected_code=0,
+            expected_output_contains=[
+                f"Flow run '{flow_run.name}' - Log {i} from flow_run {flow_run.id}."
+                for i in range(self.PAGE_SIZE)
```

Comment:
If I'm understanding this test correctly, the goal here is to create 201 logs and then return those 201 logs. Would we want this to be `self.PAGE_SIZE + 1` to match with the number of created logs? Regardless, adding `+ 1` still results in a successful test.

Reply:
That's a good catch! Indeed `self.PAGE_SIZE + 1` is the desired case.